File: sample/src/main/java/com/stepstone/stepper/sample/adapter/FormFragmentStepAdapter.java
package com.stepstone.stepper.sample.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.app.FragmentManager;

import com.stepstone.stepper.Step;
import com.stepstone.stepper.adapter.AbstractFragmentStepAdapter;
import com.stepstone.stepper.sample.step.fragment.FormStepFragment;

public class FormFragmentStepAdapter extends AbstractFragmentStepAdapter {

    public FormFragmentStepAdapter(@NonNull FragmentManager fm, @NonNull Context context) {
        super(fm, context);
    }

    @Override
    public Step createStep(int position) {
        return FormStepFragment.newInstance();
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: sample/src/main/java/com/stepstone/stepper/sample/adapter/FormFragmentStepAdapter.kt
package com.stepstone.stepper.sample.adapter

import android.content.Context
import android.support.v4.app.FragmentManager

import com.stepstone.stepper.Step
import com.stepstone.stepper.adapter.AbstractFragmentStepAdapter
import com.stepstone.stepper.sample.step.fragment.FormStepFragment

class FormFragmentStepAdapter(fm: FragmentManager, context: Context) : AbstractFragmentStepAdapter(fm, context) {

    override fun createStep(position: Int): Step {
        return FormStepFragment.newInstance()
    }

    override fun getCount(): Int {
        return 3
    }
}